src/Init/Prelude.lean

```diff
@@ -1098,6 +1098,7 @@ until `c` is known.
         | Or.inl h => hp h
         | Or.inr h => hq h
 
+@[inline]
```

Comment:
Interesting oversight, thanks for noticing this! Did you check whether there are other instances in core which could profit from an `inline`? If not I would suggest we could maybe do those changes in a separate PR where as much as possible is fixed at once.